src/modules/specializations/specializations.service.ts

```diff
@@ -35,8 +35,32 @@ export class SpecializationsService {
     }
   }
 
-  async findOneById(id: string): Promise<Specialization> {
+  async delete(id: string): Promise<boolean> {
     try {
+      const specialization = await this.findOneById(id, {
+        relations: ['doctors'],
+      });
+      if (specialization.doctors.length >= 1) {
+        throw new HttpException(
+          'You can’t remove the last Specialization because it’s still assigned to a doctor',
+          HttpStatus.BAD_REQUEST,
+        );
+      }
+      await this.specializationRepository.delete(id);
+      return true;
+    } catch (err) {
+      throw new HttpException(err, HttpStatus.BAD_REQUEST);
+    }
+  }
+
+  async findOneById(
+    id: string,
+    options?: FindOneOptions<Specialization>,
+  ): Promise<Specialization> {
+    try {
+      if (options) {
+        return await this.specializationRepository.findOne(id, options);
+      }
       return await this.specializationRepository.findOne(id);
     } catch (err) {
```

Comment:
Very strange, do you really need this method (findOneById)? Why can't you just use the repository?

Reply:
I'm using it in several modules, I'm using services instead of a data layer as Denis said

Reply:
I'm okay with that if it's used in multiple places.